src/sentry/api/serializers/models/sentry_app.py

```diff
@@ -30,10 +33,16 @@ def serialize(self, obj, attrs, user, access):
             "allowedOrigins": obj.application.get_allowed_origins(),
         }
 
+        data["featureData"] = []
+
+        if obj.status != SentryAppStatus.INTERNAL:
+            features = IntegrationFeature.objects.filter(sentry_app_id=obj.id)
+            data["featureData"] = map(lambda x: serialize(x, user), features)
+
         if obj.status == SentryAppStatus.PUBLISHED and obj.date_published:
             data.update({"datePublished": obj.date_published})
 
-        if is_active_superuser(env.request) or (
+        if (env.request and is_active_superuser(env.request)) or (
```

Comment:
@NisanthanNanthakumar You don't need parentheses around a conditional statement in python :)

Reply:
@scefali just to confirm, I want to evaluate the two `and` comparisons first and then do the `or` comparison. I think we need the parentheses to enforce the order of operations, right?

Reply:
@NisanthanNanthakumar Oh yea, I see. Nvm. On second thought, I don't think we should make this change. Instead, I think you should just update your test to mock `is_active_superuser` to return false. I think it would be good practice for you to add a mock. Note you might need to change the import to `from sentry.auth import superuser` then use `superuser. is_active_superuser` to get this to work.

Reply:
@scefali But, in other serializers where they use `is_active_superuser`, they use similar logic (which is where I got the idea from). Ex: `src/sentry/api/serializers/models/team.py` line 87.

Reply:
@NisanthanNanthakumar fair enough. You'll get some practice with mocks later.

Reply:
imo, I think the change is okay. You are going to have to update the endpoint tests which should cover both superusers and non-superusers, and either in the endpoint tests or if you, in the serializer as well, we should make sure we test the internal integrations returning `[]` for the features

Reply:
Good point about the internal integration test, definitely need that. And yes, existing tests will fail which will need to updated.